CSharp/DevVmPowershell/Helpers/Implementations/ImagingHelper.cs

```diff
@@ -158,18 +168,37 @@ private async Task<bool> JobCompletedSuccessfullyAsync(int workspaceArtifactId,
 			const int sleepTimeInMilliSeconds = Constants.Waiting.SLEEP_TIME_IN_SECONDS * 1000;
 			int currentWaitTimeInMilliseconds = 0;
 
-			Guid fieldGuid = Constants.Guids.Fields.ImagingSet.Status;
-
 			try
 			{
 				while (currentWaitTimeInMilliseconds < maxTimeInMilliseconds && jobComplete == false)
 				{
-					Thread.Sleep(sleepTimeInMilliSeconds);
-
-					RDO job = await Task.Run(() => RsapiClient.Repositories.RDO.ReadSingle(imagingSetArtifactId));
-					jobComplete = job[fieldGuid].ValueAsFixedLengthText.Contains("Completed");
-
-					currentWaitTimeInMilliseconds += sleepTimeInMilliSeconds;
+					HttpClient httpClient = RestHelper.GetHttpClient(InstanceAddress, AdminUsername, AdminPassword);

```

Comment:
Can we use the Retry function from the Polly tool which Tom used in one of the helpers to retry waiting. I believe its in Application Helper class. 
I think that approach is much cleaner and re-usable and better than any custom code we write. Also code will be consistent across our helpers.

- [ ] use the Retry function from the Polly tool

Reply:
Just a note, if the function you're retrying doesn't throw an error if you don't get the desired value, you can quickly wrap that up in a dynamic function to throw one if the value isn't want you wanted.

Reply:
Updated to use RetryLogicHelper